watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Activities/LandingPageActivity.java

```diff
@@ -190,6 +223,32 @@ public void onCompleted(GraphUser user, com.facebook.Response response) {
         }
     }
 
+    /**
+     * Registers the application with GCM servers asynchronously.
+     * <p>
+     * Stores the registration ID and app versionCode in the application's
+     * shared preferences.
+     */
+    public void registerInBackground(final HashMap<String, Object> objectParams, final String type) {
+        new AsyncTask<String, String, String>() {
```

Comment:
Not sure why this needs to be done in the background. Also, if we're registering for a user id asynchronously, then what will be the user id on the next screen when the user tries to request tasks?